transports/onion/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+#![doc(html_logo_url = "https://libp2p.io/img/logo_small.png")]
+#![doc(html_favicon_url = "https://libp2p.io/img/favicon.png")]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+
+use std::pin::Pin;
+use std::sync::Arc;
+
+use address::{dangerous_extract_tor_address, safe_extract_tor_address};
+use arti_client::{TorAddrError, TorClient, TorClientBuilder};
+use futures::{future::BoxFuture, FutureExt};
+use libp2p_core::{transport::TransportError, Multiaddr, Transport};
+use tor_rtcompat::Runtime;
+
+mod address;
+mod provider;
+
+#[doc(inline)]
+pub use provider::OnionStream;
+
+#[derive(Debug, thiserror::Error)]
+pub enum OnionError {
+    #[error("error during address translation")]
+    AddrErr(#[from] TorAddrError),
+    #[error("error in arti")]
+    ArtiErr(#[from] arti_client::Error),
+}
+
+#[derive(Clone)]
+pub struct OnionClient<R: Runtime> {
```

Comment:
```suggestion
pub struct OnionTransport<R: Runtime> {
```

This follows the naming convention from the remaining crates.